HCCDOC-3084 Add release note about the new Rapid Recommendations feature (#83273)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-insights-operator-enhancements_{context}"]
=== Insights Operator

> * The Insights Operator now collects the `haproxy_exporter_server_threshold` metric. (link:https://issues.redhat.com/browse/OCPBUGS-36687[*OCPBUGS-36687*])

Author: FYI anyone :eyes:  - This line item has actually been moved down to line 570.

// Engineering reference CCXDEV-12582

> * Previously, the Insights Operator gathered information about all Ingress Controller certificates, including their `NotBefore` and `NotAfter` dates. This data is now compiled into a `JSON` file located at `aggregated/ingress_controllers_certs.json` for easier monitoring of certificate validity across the cluster. (link:https://issues.redhat.com/browse/OCPBUGS-35727[*OCPBUGS-35727*])

Author: FYI anyone 👀 - This line item has actually been moved down to line 576.

[id="ocp-4-17-insights-operator-enhancements_{context}"]
=== Insights Operator

// Engineering reference CCXDEV-12582

> The Insights Operator now collects more {product-title} container log data from namespaces prefixed with either the  “openshift-” or “kube-” prefix and generates recommendations much faster.

Reviewer: ```suggestion
The Insights Operator now collects more {product-title} container log data from namespaces prefixed with either the  `openshift-` or `kube-` prefix and generates recommendations much faster.
```

> Enhancements have also been made to give you more flexibility in how the data to be collected gets defined for your service.

Author: If this was a procedure or a recommended action, I would agree.
The suggestion above significantly changes the technical meaning. 

While the active voice is much clearer for task-oriented content, especially when you know who is doing the action, the passive voice is recommended in certain content types like reference and prereq info. See IBM style on it.

It's also sometimes much better for prereq information like the description of features in release notes where it's more appropriate to focus on the receiver of the action/benefit. Passive voice is frequently used in these release notes.

> ==== Rapid Recommendations

Author: Closing comment on this basis. Thanks.

This release introduces a new feature called Rapid Recommendations, which provides a more dynamic and version-independent mechanism for remotely configuring the rules that determine which data the Insights Operator collects.

Rapid Recommendations builds on the existing conditional data gathering mechanism.
The Insights Operator connects to a secure remote endpoint service running on `/console.redhat.com` to retrieve definitions that contain the rules for determining which container log messages to be filtered and collected by Red Hat.

> The conditional data-gathering definitions, also referred to as rules, get configured through an attribute named `conditionalGathererEndpoint` in the link:https://github.com/openshift/insights-operator/blob/master/config/pod.yaml[`pod.yml`] configuration file.

Author: As the customer/user (e.g. SRE administrator) doesn't configure or control this attribute, the Red Hat host does, `get` is more appropriate here. Stating `are` implies anyone can, which isn't quite true.

(I wondered then why we even have to mention it... but the SME confirmed we do because the endpoint differs and offers more flexibility and backwards compatibility for customers through their subscription architecture, which they don't always directly configure/control).

```bash

> conditionalGathererEndpoint: https://console.redhat.com/api/gathering/v2/%s/gathering_rules

Reviewer: ```suggestion
[source,bash]
----
conditionalGathererEndpoint: https://console.redhat.com/api/gathering/v2/%s/gathering_rules
----
```

Reviewer: Looks like `bash` is a recognized value for the `source` attribute: https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-highlighter/#shell-vs-console

```

> NOTE: Previously, the rules for determining the data that the Insights Operator collects were hard-coded and tied to the corresponding {product-title} version.

Reviewer: ```suggestion
[NOTE]
==== 
Previously, the rules for determining the data that the Insights Operator collects were hard-coded and tied to the corresponding {product-title} version.
====
```

The preconfigured endpoint URL now provides a placeholder (`%s`) for defining a target version of {product-title}.

> 

Author: The note on line 60 is already a valid [AsciiDoc admonition](https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/) for a single paragraph. See the PR preview:

![image](https://github.com/user-attachments/assets/421c2508-f84e-4abf-9fc8-3a5c2fc829c0)

==== More data collected and recommendations added

The Insights Operator now gathers more data to detect the following scenarios, which other applications can use to generate remedial recommendations to proactively manage your {product-title} deployments:

// Engineering reference: CCXDEV-12899
* Detects pods and namespaces that use the link:https://access.redhat.com/articles/7065170[deprecated OpenShiftSDN CNI plugin] and generates a recommendation for the possible actions you should take depending on the data collected from your deployment.
// Engineering reference: OSPRH-5904
* Collects custom resource definitions (CRD) from {rh-openstack}.
// Engineering reference: CCXDEV-13001
* Collects the `haproxy_exporter_server_threshold` metric to detect the problem and remediation reported in link:https://issues.redhat.com/browse/OCPBUGS-36687[*OCPBUGS-36687*].
// Engineering reference: CCXDEV-12758

> * Collects data to detect custom Prometheus Alertmanager instances that are not in the "openshift-monitoring" namespace because they could potentially impact the management of corresponding resources.

Reviewer: ```suggestion
* Collects data to detect custom Prometheus Alertmanager instances that are not in the `openshift-monitoring` namespace because they could potentially impact the management of corresponding resources.
```

// Engineering reference: CCXDEV-12503

> * Detects the upcoming expiry of the default ingress controller expiration certificate, which other applications and services can use to generate recommendations to renew the certificate before the expiry date.

Reviewer 2: ```suggestion
* Detects the upcoming expiry of the default Ingress Controller expiration certificate, which other applications and services can use to generate recommendations to renew the certificate before the expiry date.
```

Reviewer 2: https://redhat-documentation.github.io/supplementary-style-guide/#ingress-controller

The Insights Operator connects to a secure remote endpoint service running on `/console.redhat.com` to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red Hat.

The conditional data-gathering definitions, also referred to as rules, get configured through an attribute named `conditionalGathererEndpoint` in the link:https://github.com/openshift/insights-operator/blob/master/config/pod.yaml[`pod.yml`] configuration file.

[source,bash]
----
conditionalGathererEndpoint: https://console.redhat.com/api/gathering/v2/%s/gathering_rules
----

[NOTE]
====
Previously, the rules for determining the data that the Insights Operator collects were hard-coded and tied to the corresponding {product-title} version.
====
The preconfigured endpoint URL now provides a placeholder (`%s`) for defining a target version of {product-title}.

==== More data collected and recommendations added

The Insights Operator now gathers more data to detect the following scenarios, which other applications can use to generate remedial recommendations to proactively manage your {product-title} deployments:

// Engineering reference: CCXDEV-12899

> * Detects pods and namespaces that use the link:https://access.redhat.com/articles/7065170[deprecated OpenShiftSDN CNI plugin] and generates a recommendation for the possible actions you should take depending on the data collected from your deployment.

Reviewer: ```suggestion
* Detects pods and namespaces that use the link:https://access.redhat.com/articles/7065170[deprecated OpenShift SDN CNI plugin] and generates a recommendation for the possible actions you should take depending on the data collected from your deployment.
```

Reviewer: HI @michelle-purcell . Sorry, one more change. I can then merge the PR. Thanks for your work on this update.

Reviewer 3: Hi @michelle-purcell Hi, can you please make this change and then I can merge? I looked, but I don't see it. Thank you!

// Engineering reference CCXDEV-12582

The Insights Operator now collects more {product-title} container log data from namespaces prefixed with either the  `openshift-` or `kube-` prefix and generates recommendations much faster.
Enhancements have also been made to give you more flexibility in how the data to be collected gets defined for your service.

==== Rapid Recommendations

This release introduces a new feature called Rapid Recommendations, which provides a more dynamic and version-independent mechanism for remotely configuring the rules that determine which data the Insights Operator collects.

Rapid Recommendations builds on the existing conditional data gathering mechanism.
The Insights Operator connects to a secure remote endpoint service running on `/console.redhat.com` to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red Hat.

The conditional data-gathering definitions, also referred to as rules, get configured through an attribute named `conditionalGathererEndpoint` in the link:https://github.com/openshift/insights-operator/blob/master/config/pod.yaml[`pod.yml`] configuration file.

```bash

> [source,bash]

Reviewer: Hi @michelle-purcell . JUst one change here as the `source,bash` should not show in the preview. You can drop the ```bash and ``` delimiters. This should fix this issue. 

![Screenshot from 2024-10-23 17-29-58](https://github.com/user-attachments/assets/7a9ee7e7-12a5-4c95-85cc-cbb7c2cde029)
